controllers/watchers/target_pod_test.go

```diff
@@ -0,0 +1,77 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023 Datadog, Inc.
+
+package watchers_test
+
+import (
+	chaosv1beta1 "github.com/DataDog/chaos-controller/api/v1beta1"
+	"github.com/DataDog/chaos-controller/controllers/watchers"
+	"github.com/DataDog/chaos-controller/controllers/watchers/mocks"
+	"github.com/DataDog/chaos-controller/log"
+	"github.com/DataDog/chaos-controller/metrics/noop"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+var _ = Describe("target pod", func() {
+	Describe("Handler", func() {
+		//	@ToDo
```

Comment:
Still a TODO?

Reply:
Indeed! 
IMO: this watcher is really complex and it will require a lot of reverse engineering to cover all the cases... To implement the unit tests it could more faster to pair with @clairecng and/or @nathantournant